### PR TITLE
base structure of authorization-impl module

### PR DESCRIPTION
**Pull request by AlexKazmirchuk**

### CODE REVIEW

**Comment by AlexKazmirchuk** [comment]
h_r_add_classes_br -> handlers_routers_adding_classes_branch

File: softgroup/services/authorization/authorization-api/src/main/java/com/softgroup/authorization/api/message/RegisterResponse.java [hunk 1]
```diff
@@ -8,4 +8,32 @@
  */
 public class RegisterResponse implements ResponseData {
 	private static final long serialVersionUID = -5146888202653750948L;
+
+	private Integer registrationRequestUuid;
```
Comment (Dominys) [inline softgroup/services/authorization/authorization-api/src/main/java/com/softgroup/authorization/api/message/RegisterResponse.java]: @AlexKazmirchuk String.

File: softgroup/services/authorization/authorization-api/src/main/java/com/softgroup/authorization/api/message/RegisterResponse.java [hunk 2]
```diff
@@ -8,4 +8,32 @@
...
+	private Integer registrationTimeoutSec;
+	private Integer authCode;
```
Comment (Dominys) [inline softgroup/services/authorization/authorization-api/src/main/java/com/softgroup/authorization/api/message/RegisterResponse.java]: @AlexKazmirchuk String

File: softgroup/services/authorization/authorization-api/src/main/java/com/softgroup/authorization/api/message/RegisterRequest.java [hunk 3]
```diff
@@ -8,4 +8,32 @@
  */
 public class RegisterRequest implements RequestData{
 	private static final long serialVersionUID = -645554380912935546L;
+
+	private String phoneNumber;
+	private String localeCode;
+	private Integer deviceID;
```
Comment (Dominys) [inline softgroup/services/authorization/authorization-api/src/main/java/com/softgroup/authorization/api/message/RegisterRequest.java]: @AlexKazmirchuk String

File: softgroup/services/authorization/authorization-api/src/main/java/com/softgroup/authorization/api/message/SmsConfirmRequest.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+import com.softgroup.common.protocol.RequestData;
+
+/**
+ * Created by sasha on 23.02.17.
+ */
+public class SmsConfirmRequest implements RequestData {
+    private static final long serialVersionUID = 4895237867740981751L;
+
+    private Integer authCode;
```
Comment (Dominys) [inline softgroup/services/authorization/authorization-api/src/main/java/com/softgroup/authorization/api/message/SmsConfirmRequest.java]: @AlexKazmirchuk String

File: softgroup/services/authorization/authorization-imp/src/main/java/com/softgroup/authorithation/impl/handler/LoginRequestHandler.java [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+@Component
+public class LoginRequestHandler<T extends LoginRequest, R extends LoginResponse> extends AbstractRequestHandler implements AuthorizationRequestHandler {
+
+    @Override
+    public String getName() {
+        return "login";
+    }
+
+    @Override
+    public Response handle(Request msg) {
```
Comment (Dominys) [inline softgroup/services/authorization/authorization-imp/src/main/java/com/softgroup/authorithation/impl/handler/LoginRequestHandler.java]: @AlexKazmirchuk Where is generic types???

File: softgroup/services/authorization/authorization-imp/src/main/java/com/softgroup/authorithation/impl/handler/RegisterRequestHandler.java [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+@Component
+public class RegisterRequestHandler <T extends RegisterRequest, R extends RegisterResponse> extends AbstractRequestHandler implements AuthorizationRequestHandler {
+
+    @Override
+    public String getName() {
+        return "register";
+    }
+
+    @Override
+    public Response handle(Request msg) {
```
Comment (Dominys) [inline softgroup/services/authorization/authorization-imp/src/main/java/com/softgroup/authorithation/impl/handler/RegisterRequestHandler.java]: @AlexKazmirchuk Types?

File: softgroup/services/authorization/authorization-imp/src/main/java/com/softgroup/authorithation/impl/handler/SmsConfirmRequestHandler.java [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+
+@Component
+public class SmsConfirmRequestHandler <T extends SmsConfirmRequest, R extends SmsConfirmResponse> extends AbstractRequestHandler implements AuthorizationRequestHandler {
+    @Override
+    public String getName() {
+        return "sms_confirm";
+    }
+
+    @Override
+    public Response handle(Request msg) {
```
Comment (Dominys) [inline softgroup/services/authorization/authorization-imp/src/main/java/com/softgroup/authorithation/impl/handler/SmsConfirmRequestHandler.java]: @AlexKazmirchuk Types...

File: softgroup/services/authorization/authorization-imp/src/main/java/com/softgroup/authorithation/impl/router/AuthorizationRouterHandlerImpl.java [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Created by sasha on 23.02.17.
+ */
+
+@Component
+public class AuthorizationRouterHandlerImpl extends AbstractRouterHandler implements AuthorizationRouterHandler{
```
Comment (Dominys) [inline softgroup/services/authorization/authorization-imp/src/main/java/com/softgroup/authorithation/impl/router/AuthorizationRouterHandlerImpl.java]: @AlexKazmirchuk Move all to abstract

File: softgroup/services/authorization/authorization-imp/src/main/java/com/softgroup/authorithation/impl/handler/SmsConfirmRequestHandler.java [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+import com.softgroup.common.protocol.Response;
+import com.softgroup.common.router.api.AbstractRequestHandler;
+import org.springframework.stereotype.Component;
+
+/**
+ * Created by sasha on 23.02.17.
+ */
+
+@Component
+public class SmsConfirmRequestHandler <T extends SmsConfirmRequest, R extends SmsConfirmResponse> extends AbstractRequestHandler implements AuthorizationRequestHandler {
```
Comment (Dominys) [inline softgroup/services/authorization/authorization-imp/src/main/java/com/softgroup/authorithation/impl/handler/SmsConfirmRequestHandler.java]: @AlexKazmirchuk Why you need extends in generic type here ?

**Comment by Dominys** [comment]
@AlexKazmirchuk No need to clear empty strings in pom files. It makes them less readable and make mess in PR.

Second - make smaller PRs.

File: softgroup/services/authorization/authorization-api/src/main/java/com/softgroup/authorization/api/message/RegisterResponse.java (hunk 1 above)
Comment (AlexKazmirchuk) [reply softgroup/services/authorization/authorization-api/src/main/java/com/softgroup/authorization/api/message/RegisterResponse.java]: fixed

File: softgroup/services/authorization/authorization-api/src/main/java/com/softgroup/authorization/api/message/RegisterResponse.java (hunk 2 above)
Comment (AlexKazmirchuk) [reply softgroup/services/authorization/authorization-api/src/main/java/com/softgroup/authorization/api/message/RegisterResponse.java]: fixed

File: softgroup/services/authorization/authorization-api/src/main/java/com/softgroup/authorization/api/message/RegisterRequest.java (hunk 3 above)
Comment (AlexKazmirchuk) [reply softgroup/services/authorization/authorization-api/src/main/java/com/softgroup/authorization/api/message/RegisterRequest.java]: @Dominys fixed

File: softgroup/services/authorization/authorization-api/src/main/java/com/softgroup/authorization/api/message/SmsConfirmRequest.java (hunk 4 above)
Comment (AlexKazmirchuk) [reply softgroup/services/authorization/authorization-api/src/main/java/com/softgroup/authorization/api/message/SmsConfirmRequest.java]: @Dominys fixed

File: softgroup/services/authorization/authorization-imp/src/main/java/com/softgroup/authorithation/impl/handler/LoginRequestHandler.java (hunk 5 above)
Comment (AlexKazmirchuk) [inline softgroup/services/authorization/authorization-imp/src/main/java/com/softgroup/authorithation/impl/handler/LoginRequestHandler.java]: @Dominys якщо на Request чіпляти generic то компілятор ругається . Що я не так зробив, підкажіть де помилка ? 


Якщо лише на Response то все норм

Comment (AlexKazmirchuk) [reply softgroup/services/authorization/authorization-imp/src/main/java/com/softgroup/authorithation/impl/handler/LoginRequestHandler.java]: @Dominys  Now that`s rigth ?
```java
@Component
public class LoginRequestHandler extends AbstractRequestHandler<LoginRequest,LoginResponse> implements AuthorizationRequestHandler {

    @Override
    public String getName() {
        return "login";
    }

    @Override
    public Response<LoginResponse> handle(Request<?> msg) {
        // implement later


        //
        return null;
    }
}
```

**Comment by Dominys** [comment]
@AlexKazmirchuk Split into smaller PRs !!!

**Pull request closed without merging**